### ISSUE TITLE
Support Push and Local Toast Notifications for Packaged and Unpackaged Apps

### ISSUE THREAD

**Issue by vaheeshta** [microsoft/WindowsAppSDK#1614]
# Proposal: Support Push and Local Toast Notifications for Packaged and Unpackaged Apps

## Summary
This feature's goal is to allow Windows apps to send push and local toast notifications and handle activation using a common set of APIs. 

## Rationale
* The toast notification development experience should not be dependent on packaging. Both packaged and unpackaged apps should be able to register, send, and activate toast notifications with common APIs. Push toast notifications should especially be supported for unpackaged apps.
* Unpackaged apps should not need a Start menu shortcut to use toast notifications. 
* Toast notifications should be actionable. We should make it easier for developers of any Windows app to properly support activation.

## Scope
| Capability | Priority |
| :---------- | :------- |
|All Windows apps can register and display toast notifications with a common set of APIs | Must |
| Windows apps can receive push toast notifications | Must |
| Support foreground and background activation of toasts | Must |
| Windows apps can use Windows Community Toolkit to construct toast content | Must |
| Windows apps can manage toast notification history | Must |
| Windows apps can construct toast content with content functionality including visuals, actions, audio, adaptive content, progress bar, etc. | Must |
| Windows apps should be able to query settings and policies to see if Toasts are disabled on the local system | Must |
| Windows apps can schedule toasts | Should |
| Windows apps can use toast collections | Could |

Note: We will bring new NotificationBuilder toast content builders to a later WinAppSDK release while we prioritize the above capabilities. Apps can continue to use the Windows Community Toolkit builder in the meantime. 

### API Proposal
| Class | Method | Description|
| :---------- | :------- |  :------- |
|ToastNotificationManager|CreateToastNotifier() |Creates and initializes a new instance of the ToastNotification, bound to the calling application, which lets you raise a toast notification to that app.|
| ToastNotificationManager |GetToastCollectionManager() |Creates a ToastCollectionManager that you can use to save, update, and clear notification groups. |
|ToastNotificationManager |CreateToastNotifierForToastCollectionAsync(String collectionId) |Creates and initializes a new instance of the ToastNotifier that lets you raise a toast notification within the specified toast collection.|
| ToastNotificationManager|GetHistoryForToastCollectionAsync(String collectionId) |Gets the notification history for the specified toast collection. |
|ToastNotifier | Show(ToastNotification)| Displays the specified toast notification.|
|ToastNotifier |RemoveByIdentifier(Uint toastId) | Removes the specified toast notification from the screen.|
|ToastNotifier |Update(NotificationData, String) | Updates the existing toast notification that has the specified tag.|
|ToastNotifier |Update(NotificationData, String tag, String group)|Updates the existing toast notification that has the specified tag. |
|ToastNotifier | GetScheduledToastNotifications()|Gets the collection of ScheduledToastNotification objects that this app has scheduled for display. |
|ToastNotifier | AddToSchedule(ScheduledToastNotification)| Adds a ScheduledToastNotification for later display by Windows.|
| ToastNotifier |RemoveFromSchedule(ScheduledToastNotification)|Cancels the scheduled display of a specified ScheduledToastNotification.|
|ToastNotificationHistory |RemoveAll() |Removes all notifications sent by this app from action center. |
|ToastNotificationHistory |GetHistoryAsync() | Gets the collection of toasts currently in Action Center.|
|ToastNotificationHistory |Remove(String tag) |Removes an individual toast, with the specified tag label, from action center.|
|ToastNotificationHistory |Remove(String tag, String group) | Removes a toast notification from the action using the notification's tag and group labels.|
| ToastNotificationHistory| RemoveGroup(String group) | Removes a group of toast notifications, identified by the specified group label, from action center.|
|ToastCollectionManager |GetAllToastCollectionsAsync() | Starts the retrieval of all toast notification groups created by this manager.|
|ToastCollectionManager |GetToastCollectionAsync(String) |Starts the retrieval of the specified notification group. |
|ToastCollectionManager |RemoveAllToastCollectionsAsync() | Starts the asynchronous removal of all toast notification groups for the app.|
|ToastCollectionManager |RemoveToastCollectionAsync(String) |Starts the asynchronous removal of the specified toast notification group.|
|ToastCollectionManager |SaveToastCollectionAsync(ToastCollection) | Starts the asynchronous creation or update of the specified toast notification group.|

**Comment by shelllet** [comment]
`Windows::UI::Notifications::ToastNotificationManager::CreateToastNotifier().Show(notification);`
```
Exception thrown at 0x00007FFC7E49466C (KernelBase.dll) in Workflow.exe: WinRT originate error - 0x80070490 : 'Element not found.'.
```

**Comment by vaheeshta** [comment]
Support for toast notifications for packaged and unpackaged apps has been implemented with #2192
